### PR TITLE
Project Phoenix: `net.minecraft.world.level.block` and children

### PR DESCRIPTION
**Pull request by diesieben07**
Only obvious things are named / corrected from MCP. No new Javadoc except small changes from MCP for names/wording.
If something is wrong but unclear, I've simply removed it, it will be added back later.

I have also removed any copied Javadoc from overridden methods so that IDEs can simply display the inherited one, instead of us having to maintain the copy-pasted version.

### CODE REVIEW

File: data/net/minecraft/world/level/block/state/predicate/BlockMaterialPredicate.mapping [hunk 1]
```diff
@@ -1,3 +1,7 @@
 CLASS net/minecraft/world/level/block/state/predicate/BlockMaterialPredicate
+	METHOD <init> (Lnet/minecraft/world/level/material/Material;)V
+		ARG 1 material
 	METHOD forMaterial (Lnet/minecraft/world/level/material/Material;)Lnet/minecraft/world/level/block/state/predicate/BlockMaterialPredicate;
 		ARG 0 material
+	METHOD test (Ljava/lang/Object;)Z
+		ARG 1 state
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/block/state/predicate/BlockMaterialPredicate.mapping]: Please remove this method.

File: data/net/minecraft/world/level/block/state/predicate/BlockPredicate.mapping [hunk 2]
```diff
@@ -1,3 +1,7 @@
 CLASS net/minecraft/world/level/block/state/predicate/BlockPredicate
+	METHOD <init> (Lnet/minecraft/world/level/block/Block;)V
+		ARG 1 block
 	METHOD forBlock (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/world/level/block/state/predicate/BlockPredicate;
-		ARG 0 blockType
+		ARG 0 block
+	METHOD test (Ljava/lang/Object;)Z
+		ARG 1 state
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/block/state/predicate/BlockPredicate.mapping]: Please remove this method.

File: data/net/minecraft/world/level/block/state/predicate/BlockStatePredicate.mapping [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
 	METHOD applies (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/state/properties/Property;Ljava/util/function/Predicate;)Z
-		ARG 1 blockState
+		ARG 1 state
 		ARG 2 property
-		ARG 3 predicate
+		ARG 3 valuePredicate
 	METHOD forBlock (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/world/level/block/state/predicate/BlockStatePredicate;
 		ARG 0 block
+	METHOD test (Ljava/lang/Object;)Z
+		ARG 1 state
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/block/state/predicate/BlockStatePredicate.mapping]: Please remove this method.

**Review by sciwhiz12 — DISMISSED** [review]
Please remove the following methods for the automatic data validation task to pass.

To my understanding, somehow Enigma is mapping the synthetic bouner counterpart of the method instead of the actual method (as evidenced by the descriptor referencing `java/lang/Object`). 

I am working on finding the cause and fixing it, but for now to fix it either manually remove the methods from the files or run in order the following tasks: `createStagingFromInputs sanitizeStagingData promoteStangingToProduction` (or more succintly, `cSFI ssD pSTP` on the command line).

File: data/net/minecraft/world/level/block/KelpBlock.mapping [hunk 4]
```diff
@@ -7,8 +7,7 @@ CLASS net/minecraft/world/level/block/KelpBlock
 		ARG 3 state
 		ARG 4 fluid
 	METHOD getBlocksToGrowWhenBonemealed (Ljava/util/Random;)I
-		COMMENT Used to determine how much to grow the plant when using bonemeal. Kelp always returns 1, where as the nether vines return a random value at least 1.
-		ARG 1 random
+		ARG 1 rand
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/KelpBlock.mapping]: `random`

File: data/net/minecraft/world/level/block/SoulSandBlock.mapping [hunk 5]
```diff
@@ -30,9 +30,6 @@ CLASS net/minecraft/world/level/block/SoulSandBlock
 		ARG 3 pos
 		ARG 4 rand
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/SoulSandBlock.mapping]: `random`

File: data/net/minecraft/world/level/block/SugarCaneBlock.mapping [hunk 6]
```diff
@@ -22,9 +21,6 @@ CLASS net/minecraft/world/level/block/SugarCaneBlock
 		ARG 3 pos
 		ARG 4 rand
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/SugarCaneBlock.mapping]: `random`

File: data/net/minecraft/world/level/block/VineBlock.mapping [hunk 7]
```diff
@@ -37,25 +39,17 @@ CLASS net/minecraft/world/level/block/VineBlock
...
-		COMMENT Performs a random tick on a block.
 		ARG 1 state
 		ARG 2 level
 		ARG 3 pos
 		ARG 4 random
 	METHOD rotate (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/Rotation;)Lnet/minecraft/world/level/block/state/BlockState;
-		COMMENT Returns the blockstate with the given rotation from the passed blockstate. If inapplicable, returns the passed blockstate.
-		COMMENT @deprecated call via {@link IBlockState#withRotation(Rotation)} whenever possible. Implementing/overriding is fine.
 		ARG 1 state
 		ARG 2 rot
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/VineBlock.mapping]: `rotation`

File: data/net/minecraft/world/level/block/WallBlock.mapping [hunk 8]
```diff
@@ -36,23 +36,22 @@ CLASS net/minecraft/world/level/block/WallBlock
...
 		ARG 2 mirror
 	METHOD propagatesSkylightDown (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/BlockGetter;Lnet/minecraft/core/BlockPos;)Z
 		ARG 1 state
 		ARG 2 reader
 		ARG 3 pos
 	METHOD rotate (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/Rotation;)Lnet/minecraft/world/level/block/state/BlockState;
-		COMMENT Returns the blockstate with the given rotation from the passed blockstate. If inapplicable, returns the passed blockstate.
-		COMMENT @deprecated call via {@link IBlockState#withRotation(Rotation)} whenever possible. Implementing/overriding is fine.
 		ARG 1 state
 		ARG 2 rot
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/WallBlock.mapping]: `rotation`

File: data/net/minecraft/world/level/block/grower/SpruceTreeGrower.mapping [hunk 9]
```diff
@@ -1,8 +1,6 @@
 CLASS net/minecraft/world/level/block/grower/SpruceTreeGrower
 	METHOD getConfiguredFeature (Ljava/util/Random;Z)Lnet/minecraft/world/level/levelgen/feature/ConfiguredFeature;
-		COMMENT Get a {@link net.minecraft.world.gen.feature.ConfiguredFeature} of tree
 		ARG 1 random
 		ARG 2 largeHive
 	METHOD getConfiguredMegaFeature (Ljava/util/Random;)Lnet/minecraft/world/level/levelgen/feature/ConfiguredFeature;
-		COMMENT Get a {@link net.minecraft.world.gen.feature.ConfiguredFeature} of the huge variant of this tree
 		ARG 1 rand
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/grower/SpruceTreeGrower.mapping]: `random`

File: data/net/minecraft/world/level/block/state/BlockBehaviour.mapping [hunk 10]
```diff
@@ -153,13 +159,13 @@ CLASS net/minecraft/world/level/block/state/BlockBehaviour
 		ARG 4 random
 	METHOD rotate (Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/Rotation;)Lnet/minecraft/world/level/block/state/BlockState;
 		COMMENT Returns the blockstate with the given rotation from the passed blockstate. If inapplicable, returns the passed blockstate.
-		COMMENT @deprecated call via {@link IBlockState#withRotation(Rotation)} whenever possible. Implementing/overriding is fine.
+		COMMENT @deprecated call via {@link net.minecraft.world.level.block.state.BlockBehavior.BlockStateBase#rotate} whenever possible. Implementing/overriding is fine.
 		ARG 1 state
-		ARG 2 rotation
+		ARG 2 rot
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/state/BlockBehaviour.mapping]: `rotation`

File: data/net/minecraft/world/level/block/state/BlockBehaviour.mapping [hunk 11]
```diff
@@ -328,12 +353,12 @@ CLASS net/minecraft/world/level/block/state/BlockBehaviour
 			ARG 1 reader
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/block/state/BlockBehaviour.mapping]: `level`

**Review by alcatrazEscapee — CHANGES_REQUESTED** [review]

**Review by alcatrazEscapee — APPROVED** [review]

**Review by Lanse505 — APPROVED** [review]

**Comment by Lanse505** [comment]
Looks good to me, also jfc that's a lot of files to go through 😓

**Pull request merged**